airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -384,39 +384,38 @@ public WebBackendConnectionRead webBackendUpdateConnectionNew(final WebBackendCo
       throws ConfigNotFoundException, IOException, JsonValidationException {
     final List<UUID> operationIds = updateOperations(webBackendConnectionUpdate);
     final ConnectionUpdate connectionUpdate = toConnectionUpdate(webBackendConnectionUpdate, operationIds);
-
     final UUID connectionId = webBackendConnectionUpdate.getConnectionId();
-
     final ConfiguredAirbyteCatalog existingConfiguredCatalog =
         configRepository.getConfiguredCatalogForConnection(connectionId);
-    final io.airbyte.protocol.models.AirbyteCatalog existingCatalog = CatalogHelpers.configuredCatalogToCatalog(existingConfiguredCatalog);
-    final AirbyteCatalog apiExistingCatalog = CatalogConverter.toApi(existingCatalog);
-    final AirbyteCatalog newAirbyteCatalog = webBackendConnectionUpdate.getSyncCatalog();
-    final CatalogDiff catalogDiff = connectionsHandler.getDiff(apiExistingCatalog, newAirbyteCatalog);
-
-    final List<StreamDescriptor> apiStreamsToReset = getStreamsToReset(catalogDiff);
-    List<io.airbyte.protocol.models.StreamDescriptor> streamsToReset =
-        apiStreamsToReset.stream().map(ProtocolConverters::streamDescriptorToProtocol).toList();
-
     ConnectionRead connectionRead;
     connectionRead = connectionsHandler.updateConnection(connectionUpdate);
```

Comment:
In its current state, this PR will reintroduce the bug that Benoit called out, because this is now saving the updated connection before we calculate the diff

Reply:
Maybe you could add a verify InOrder to the tests to catch this in our unit tests, to prevent anyone from reintroducing this bug in the future

Reply:
I moved the fetch of the existing catalog before the save though, so I thought that would take care of that?

Reply:
Ah, I missed that, my bad. This should be fine then 👍